packages/evm/src/fungible.ts

```diff
@@ -207,4 +215,18 @@ class EvmFungibleAssetTransfer extends BaseTransfer {
 
     return createTransactionRequest(transferTx);
   }
+
+  async verifyAccountBalance(fee: EvmFee): Promise<void> {
+    const provider = new Web3Provider(this.sourceNetworkProvider);
+    const erc20 = ERC20__factory.connect(this.resource.address, provider);
+    const balance = await erc20.balanceOf(this.sourceAddress);
+
+    // TODO: check cost calculation
+    let totalCost: BigNumber = BigNumber.from(this.amount);
+    if (fee.type === FeeHandlerType.PERCENTAGE) {
+      totalCost = totalCost.add(BigNumber.from(fee.fee));
+    }
```

Comment:
@saadjhk as we deduct the fee from the amount:
```ts
this._amount = calculateAdjustedAmount(this, fee);
```

To calculate the required balance I add the fee back to calculate the initial required amount to perform the transfer.

Reply:
@Lykhoyda should we not implement this feature for `basic` and `twap` type fee calculation?

Reply:
> @saadjhk as we deduct the fee from the amount:
> 
> ```ts
> this._amount = calculateAdjustedAmount(this, fee);
> ```
> 
> To calculate the balance I add the fee back to calculate the initial required amount to perform the transfer.

I didn't get this, what do you mean by add the fee back

Reply:
@saadjhk When we call calculateAdjustedAmount method in setAmount

```ts
async setAmount(amount: bigint): Promise<void> {
    this._amount = amount;
    const fee = await this.getFee();
    this._amount = calculateAdjustedAmount(this, fee);
  }
  ```
If the fee type is a percentage, then we subtract the fee from the "amount" specified by the user. But your account should have the balance with the same amount that you set when creating a transfer (without subtracting the fee).

Reply:
Maybe it makes sense to have two variables with amount:
transferAmount ( original value)
amountWithFee (after calculateAdjustedAmount call)

Reply:
`specifiedAmount` and `adjustedAmount`